Source/Extensions/Blazorise.FluentValidation/FluentValidationHandler.cs

```diff
@@ -0,0 +1,117 @@
+﻿#region Using directives
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentValidation;
+using FluentValidation.Internal;
+using FluentValidation.Results;
+using Microsoft.Extensions.DependencyInjection;
+#endregion
+
+namespace Blazorise.FluentValidation
+{
+    public class FluentValidationHandler : IValidationHandler
+    {
+        #region Members
+
+        private readonly IServiceProvider serviceProvider;
+
+        #endregion
+
+        #region Constructors
+
+        public FluentValidationHandler( IServiceProvider serviceProvider )
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        #endregion
+
+        #region Methods
+
+        public void Validate( IValidation validation, object newValidationValue )
+        {
+            try
+            {
+                var validator = CreateValidator( validation );
+
+                validation.NotifyValidationStarted();
+                var context = CreateContext( validation );
+                var result = validator.Validate( context );
+
+                NotifyValidationChanged( result, validation );
+            }
+            catch ( Exception exc )
+            {
+                HandleException( exc, validation );
+            }
+        }
+
+        public async Task ValidateAsync( IValidation validation, object newValidationValue, CancellationToken cancellationToken = default )
+        {
+            try
+            {
+                var validator = CreateValidator( validation );
+
+                validation.NotifyValidationStarted();
+                var context = CreateContext( validation );
+                var result = await validator.ValidateAsync( context );
+
+                NotifyValidationChanged( result, validation );
+            }
+            catch ( Exception exc )
+            {
+                HandleException( exc, validation );
+            }
+        }
+
+        private void HandleException( Exception exception, IValidation validation )
+        {
+            var msg = $"An unhandled exception occurred when validating field name: '{validation.FieldIdentifier.FieldName}'";
```

Comment:
This seems like stuff applicable to standard blazor validation? EditForm, field identifiers, etc, is this applicable to our validation system?

Reply:
In this version we're only doing validation based on supplied `Model`. Similar to data annotations. TBH I'm not sure how other validators would work with fluent validation.

Reply:
Ahhh I had forgotten we had this boilerplate already from adding data annotations and stuff.